klib/tun.c

```diff
@@ -305,11 +311,14 @@ closure_function(1, 2, sysreturn, tun_ioctl,
             ip4_addr_t ipaddr = (ip4_addr_t){0};
             ip4_addr_t netmask = (ip4_addr_t){0};
             boolean bringup = false;
-            get_tun_config(tun->netif.name, &ipaddr, &netmask, &bringup);
+            u64 mtu = 0;
+            get_tun_config(tun->netif.name, &ipaddr, &netmask, &mtu, &bringup);
             netif_add(&tun->netif, &ipaddr, &netmask, &ipaddr, tun, tun_if_init, netif_input);
             netif_name_cpy(ifreq->ifr_name, &tun->netif);
             list_init(&tun->files);
             tun->next_tx = tf;
+            if (mtu > 0 && mtu < U64_FROM_BIT(16))
```

Comment:
The check against U64_FROM_BIT(16) is already done in get_tun_config(), so there is no need to redo it here.